util/ecs/ecs.go

```
@@ -0,0 +1,46 @@
+package ecs
+
+import (
+	"errors"
+
+	agentpayload "github.com/DataDog/agent-payload/gogen"
+	agentecs "github.com/DataDog/datadog-agent/pkg/metadata/ecs"
+	log "github.com/cihub/seelog"
+)
+
+var (
+	ErrECSNotAvailable = errors.New("ecs not available")
+	lastErr            string
+	globalECSUtil      *ecsUtil
+)
+
+// InitKubeUtil initializes a global kubeUtil used by later function calls.
+func InitECSUtil() error {
+	_, err := agentecs.GetPayload()
+	if err != nil && agentecs.IsAgentNotDetected(err) {
+		return ErrECSNotAvailable
+	}
+
+	globalECSUtil = &ecsUtil{}
+	return nil
+}
+
+// GetMetadata returns the metadata from the local ECS agent if available.
+func GetMetadata() *agentpayload.ECSMetadataPayload {
+	if globalECSUtil != nil {
+		return globalECSUtil.getMetadata()
+	}
+	return nil
+}
+
+type ecsUtil struct{}
+
+func (e *ecsUtil) getMetadata() *agentpayload.ECSMetadataPayload {
+	payload, err := agentecs.GetPayload()
+	if err != nil && err.Error() != lastErr {
```

Yeah we're logging and swallowing the error on purpose. I'll add a note on that